compiler-core/src/type_/tests/warnings.rs

```diff
@@ -2419,3 +2419,83 @@ pub fn main(a) {
 ",
     );
 }
+
+#[test]
+fn javascript_unsafe_int() {
+    assert_js_warning!(
+        r#"
+pub fn go() {
```

Comment:
Split these into multiple tests please, each test should only test one case.

Reply:
Have made the tests more granular